### PR TITLE
Allow flush to be sync

### PR DESCRIPTION
**Pull request by nacjacks**
Allows the flush method exposed via the SDK to be called with an optional parameter to perform the flush synchronously.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@
 *.suo
 *.user
 *.sln.docstates
+.idea/
```
Comment (KamilSzostak) [inline .gitignore]: is this required?

File: JavaScript/JavaScriptSDK/AppInsights.ts [hunk 2]
```diff
@@ -407,10 +407,19 @@ module Microsoft.ApplicationInsights {
 
         /**
          * Immediately send all queued telemetry.
+         * @param {boolean} async - If flush should be call asynchronously
          */
-        public flush() {
+        public flush(async?: boolean) {
```
Comment (KamilSzostak) [inline JavaScript/JavaScriptSDK/AppInsights.ts]: Can you use `public flush(async = true)`? Then you don't have to check if async is boolean. 

The same improvement can be done in the [sender.triggerSend()](https://github.com/Microsoft/ApplicationInsights-JS/blob/2745ffadaa834ad7532182e5aa4417b1cb7907c8/JavaScript/JavaScriptSDK/Sender.ts#L256) method. 

Thanks!

**Comment by KamilSzostak** [comment]
Thanks for contributing. This PR is addressing an issue mentioned in #445.

File: .gitignore (hunk 1 above)
Comment (nacjacks) [inline .gitignore]: Not specifically, but for some reason Webstorm decided to add that folder my project.

**Comment by jerms0518** [comment]
@KamilSzostak Any idea when this change will be released?

**Pull request merged**